src/features/courses/course-repository.ts

```diff
@@ -263,6 +263,39 @@ class CourseRepository {
         }
     }
 
+    async extendTopicByUser(options: ExtendTopicForUserOptions): Promise<UpsertResult<StudentTopicOverride>> {
+        try {
+            const found = await StudentTopicOverride.findOne({where: {...options.where, active: true}});
+
+            if (!found) {
+                const newExtension = await StudentTopicOverride.create({...options.where, ...options.updates}, {validate: true});
+
+                return {
+                    createdNewEntry: true,
+                    updatedCount: 1,
+                    updatedRecords: [newExtension],
+                };
+            } else {
+                const updates = await StudentTopicOverride.update(options.updates, {
+                    where: {
+                        ...options.where,
+                        active: true
+                    },
+                    returning: true,
+                });
+
+                return {
+                    createdNewEntry: false,
+                    updatedCount: updates[0],
+                    updatedRecords: updates[1],
+                };
+            }
+        } catch (e) {
+            this.checkCourseTopicError(e);
```

Comment:
Is this the correct check here? Doesn't this method check for topic constraint errors rather than override errors?

Reply:
You're right, I missed that when I copied and modified the original function.